modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java

```diff
@@ -158,6 +159,18 @@ public List<Map<String, String>> getObjectFieldBusinessTypeMaps(
 			));
 	}
 
+	public boolean isReadOnly(ObjectDefinition objectDefinition)
+		throws PortalException {
+
+		if(objectDefinition.isSystem()) {
+			return true;
+		}
+
```

Comment:
code DRY. should reference from BaseObjectDefinitionsDisplayContext hasUpdateObjectDefinitionPermission